src/ape/utils/trace.py

```diff
@@ -21,9 +21,11 @@
 
 from ape.exceptions import ContractError, DecodingError
 from ape.utils import ManagerAccessMixin
-from ape.utils.abi import Struct, parse_type
+from ape.utils.abi import Struct
 from ape.utils.misc import ZERO_ADDRESS
 
+from ape_ethereum.ecosystem import _parse_type
```

Comment:
This line is creating the circular import error you're getting. While we're still discussing the refactor, let's move this so that it's not erroring out. See the comment below.

Reply:
the idea of this ticket is to move the logic from parse_type into `ape_ethereum` and have it happen when calling `decode_primitive_value()` and then it does not need to be imported here.

Reply:
Yeah, this was just a temporary step to get his tests passing so he could research what else needed to change. 👍